sql/hive/src/main/java/org/apache/spark/sql/hive/mapred/CombineSplitInputFormat.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.hive.mapred;
+
+import java.io.IOException;
+import java.util.*;
+
+import com.clearspring.analytics.util.Lists;
+import com.google.common.collect.Maps;
+import com.google.common.collect.Sets;
+import org.apache.hadoop.mapred.*;
+
+public class CombineSplitInputFormat<K, V> implements InputFormat<K, V> {
```

Comment:
I think it's not strictly tie up with a specific Hive/Hadoop version. Just borrow some idea from `CombineFileInputformat` from Hadoop version 2.2.0. 
And this PR share a similar idea with https://github.com/apache/spark/pull/10572, which try to create a new proxy inputformat only that there's little bit diff with the combination logic. 
Will try to clean this code a bit.